source/opt/debug_info_manager.h

```diff
@@ -191,9 +198,9 @@ class DebugInfoManager {
   // operand is the function.
   std::unordered_map<uint32_t, Instruction*> fn_id_to_dbg_fn_;
 
-  // Mapping from local variable ids to DebugDeclare instructions whose
-  // operand is the local variable.
-  std::unordered_map<uint32_t, std::vector<Instruction*>> var_id_to_dbg_decl_;
+  // Mapping from variable or value ids to DebugDeclare or DebugValue
+  // instructions whose operand is the variable or value.
+  std::unordered_map<uint32_t, std::list<Instruction*>> var_id_to_dbg_decl_;
```

Comment:
Why did you change to a list?

Reply:
unorder_set is fine.  Avoid std::list, it is known to have poor implementations.  In general, if you know things will be small, you can never go wrong with std::vector.  It has so much less overhead than the other containers, that in practice it can still be faster.

>  std::vector::erase calls the destructor and the constructor O(N) times

A pointer does not have a constructor or destructor, so it will not be called.

Reply:
I see. Thank you for the explanation!